zerver/lib/push_notifications.py

```diff
@@ -1566,3 +1567,17 @@ def msg_format() -> str:
         # This message is not read by any of the client apps, just potentially displayed
         # via server administration tools, so it doesn't need translations.
         return "A server with hostname {hostname} already exists"
+
+
+class PushDeviceInfoDict(TypedDict):
+    status: Literal["active", "pending", "failed"]
+    error_code: str | None
+
+
+def get_push_devices(user_profile: UserProfile) -> dict[str, PushDeviceInfoDict]:
+    rows = PushDevice.objects.filter(user=user_profile)
```

Comment:
So I think we can do an optimization here of returning `{}` early if `!push_notifications_configured()`. Will save a cheap database query when that's relevant, which is probably not very often, but maybe worth doing for completeness.